packages/gitgraph-core/src/commit.ts

```diff
@@ -0,0 +1,174 @@
+import { CommitStyle } from "./template";
+import Branch from "./branch";
+
+export interface CommitOptions {
+  author: string;
+  subject: string;
+  style: CommitStyle;
+  body?: string;
+  notes?: string;
+  refs?: string[];
+  tree?: string;
+  hash?: string;
+  parents?: string[];
+  innerText?: string;
+}
+
+/**
+ * Generate a random hash.
+ *
+ * @return {string} hex string with 40 chars
+ */
+const getRandomHash = () =>
+  (
+    Math.random()
+      .toString(16)
+      .substring(3) +
+    Math.random()
+      .toString(16)
+      .substring(3) +
+    Math.random()
+      .toString(16)
+      .substring(3) +
+    Math.random()
+      .toString(16)
+      .substring(3)
+  ).substring(0, 40);
+
+export class Commit {
+  /**
+   * Ref names (injected by Gitgraph.withRefsAndTags)
+   */
+  public refs: Array<Branch["name"] | "HEAD"> = [];
+  /**
+   * Commit x position (injected by Gitgraph.withPosition)
+   */
+  public x: number = 0;
+  /**
+   * Commit y position (injected by Gitgraph.withPosition)
+   */
+  public y: number = 0;
+  /**
+   * Commit hash
+   */
+  public hash: string;
+  /**
+   * Abbreviated commit hash
+   */
+  public hashAbbrev: string;
+  /**
+   * Tree hash
+   */
+  public tree: string;
+  /**
+   * Abbreviated tree hash
+   */
+  public treeAbbrev: string;
+  /**
+   * Parent hashes
+   */
+  public parents: Array<Commit["hash"]>;
+  /**
+   * Abbreviated parent hashed
+   */
+  public parentsAbbrev: Array<Commit["hashAbbrev"]>;
+  /**
+   * Author
+   */
+  public author: {
+    /**
+     * Author name
+     */
+    name: string;
+    /**
+     * Author email
+     */
+    email: string;
+    /**
+     * Author date
+     */
+    timestamp: number;
+  };
+  /**
+   * Committer
+   */
+  public committer: {
+    /**
+     * Commiter name
+     */
+    name: string;
+    /**
+     * Commiter email
+     */
+    email: string;
+    /**
+     * Commiter date
+     */
+    timestamp: number;
+  };
+  /**
+   * Subject
+   */
+  public subject: string;
+  /**
+   * Body
+   */
+  public body: string;
+  /**
+   * Notes
+   */
+  public notes: string;
+  /**
+   * Style
+   */
+  public style: CommitStyle;
+  /**
+   * Inner text
+   */
+  public innerText?: string;
+  /**
+   * List of branches attached (injected by Gitgraph.withBranches)
+   */
+  public branches?: Array<Branch["name"]>;
```

Comment:
Another discussion about that is: could we have actual `Branch` references instead of just `Branch["name"]`?

Having `Branch`, we could reach for the `name` easily. Having `Branch["name"]` makes us having to find the reference from some map at our disposal, which is more painful. Also, I feel we mostly want to retrieve the actual branch when we get to this point.

But I may be missing something here. What do you think?

P.S. same question for `parents` (having the `Commit` instead of `Commit["hash"]`

Reply:
The initial idea was to mimic the output of `git2json`, since we will probably add a real `gitgraph.import` method, I think that we can refactor this part to simplify everything ;)

Another idea is to link branch and commit with `Map` like our tags pattern. Every pattern works here, we just need to choose "the best one" ^^